### PR TITLE
add documentation task

### PR DESCRIPTION
**Pull request by dduugg**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -2,6 +2,7 @@
 client/build.js
 
 ### JSDoc ###
+doc
```
Comment (dduugg) [inline .gitignore]: output directory of jsdoc grunt task

File: .jscsrc [hunk 2]
```diff
@@ -23,7 +23,6 @@
     "beforeOpeningRoundBrace": true
   },
   "disallowTrailingComma": true,
-  "disallowTrailingWhitespace": true,
```
Comment (dduugg) [inline .jscsrc]: this is already in the google preset :disappointed:

File: package.json [hunk 3]
```diff
@@ -6,7 +6,7 @@
   "scripts": {
     "start": "node index.js",
     "test": "grunt test",
-    "test-travis": "istanbul cover node_modules/mocha/bin/_mocha --report lcovonly -- --reporter spec --check-leaks test/**/*.js"
+    "test-travis": "grunt style; istanbul cover node_modules/mocha/bin/_mocha --report lcovonly -- --reporter spec --check-leaks test/**/*.js"
```
Comment (dduugg) [inline package.json]: Travis wasn't checking for style violations :sob:

File: server/schemas/schemaModel.js [hunk 4]
```diff
@@ -73,9 +73,9 @@ module.exports = {
   },
 
   /**
-   * getSchema returns the schema template.
-   * @param  {String} Schema name to inject
-   * @param  {Function} Callback to call later
+   * Get the template for the named schema.
+   * @param  {String} schemaName the name of the schema
+   * @param  {Function} callback the callback
```
Comment (dduugg) [inline server/schemas/schemaModel.js]: fixing invalid jsdoc

File: server/users/userController.js [hunk 5]
```diff
@@ -51,8 +51,6 @@ module.exports = {
     }
     User.model.findOneAsync({username: newUser})
       .then(function(user) {
-        //enable the following log statement for troubleshooting purposes
-        //console.debug("findOneAsync args:", arguments);
```
Comment (dduugg) [inline server/users/userController.js]: node doesn't have `console.debug`

File: server/schemas/schemaModel.js (hunk 4 above)
Comment (owen-d) [inline server/schemas/schemaModel.js]: :+1:

**Comment by owen-d** [comment]
:+1:  nice work

**Comment by rachelsison** [comment]
:+1:

File: Gruntfile.js [hunk 6]
```diff
@@ -46,5 +51,7 @@ module.exports = function(grunt) {
   require("load-grunt-tasks")(grunt);
   grunt.registerTask("default", ["jshint", "jscs", "mochaTest", "browserify"]);
   grunt.registerTask("build", ["browserify"]);
+  grunt.registerTask("doc", ["jsdoc"]);
+  grunt.registerTask("style", ["jshint", "jscs"]);
```
Comment (cmperez) [inline Gruntfile.js]: :clap:

**Comment by cmperez** [comment]
:+1:

File: package.json (hunk 3 above)
Comment (zindlerb) [inline package.json]: :finnadie:

**Comment by zindlerb** [comment]
:+1:

**Pull request merged**